tests/python/gpu/test_gluon_model_zoo.py

```diff
@@ -0,0 +1,163 @@
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import print_function
+import mxnet as mx
+import numpy as np
+import copy
+from mxnet import autograd
+from mxnet.gluon.model_zoo.vision import get_model
+from mxnet.test_utils import assert_almost_equal
+import sys
+
+def eprint(*args, **kwargs):
+    print(*args, file=sys.stderr, **kwargs)
+
+VAL_DATA='data/val-5k-256.rec'
+def download_data():
+    return mx.test_utils.download(
+        'http://data.mxnet.io/data/val-5k-256.rec', VAL_DATA)
+
+def test_inference():
+    all_models = ['resnet50_v1', 'vgg19_bn', 'alexnet', #'inceptionv3',
+                  'densenet201', 'squeezenet1.0', 'mobilenet0.25']
+
+    batch_size = 10
+    download_data()
+    for model_name in all_models:
+        eprint('testing inference on %s'%model_name)
+
+        data_shape = (3, 224, 224) if 'inception' not in model_name else (3, 299, 299)
+        dataIter = mx.io.ImageRecordIter(
+            path_imgrec        = VAL_DATA,
+            label_width        = 1,
+            preprocess_threads = 1,
+            batch_size         = batch_size,
+            data_shape         = data_shape,
+            label_name         = 'softmax_label',
+            rand_crop          = False,
+            rand_mirror        = False)
+        data_batch = dataIter.next()
+        data = data_batch.data[0]
+        label = data_batch.label[0]
+        gpu_data = data.as_in_context(mx.gpu())
+        gpu_label = label.as_in_context(mx.gpu())
+
+        # This is to create a model and run the model once to initialize
+        # all parameters.
+        cpu_model = get_model(model_name)
+        cpu_model.collect_params().initialize(ctx=mx.cpu())
+        cpu_model(mx.nd.array(data, ctx=mx.cpu()))
+        gpu_model = get_model(model_name)
+        gpu_model.collect_params().initialize(ctx=mx.gpu())
+        gpu_model(mx.nd.array(data, ctx=mx.gpu()))
+
+        # Force the two models have the same parameters.
+        cpu_params = cpu_model.collect_params()
+        gpu_params = gpu_model.collect_params()
+        for k in cpu_params.keys():
+            k = k.replace(cpu_params.prefix, '')
+            cpu_param = cpu_params.get(k)
+            gpu_param = gpu_params.get(k)
+            gpu_param.set_data(cpu_param.data().as_in_context(mx.gpu()))
+
+        # Run inference.
+        with autograd.record(train_mode=False):
+            cpu_out = cpu_model(mx.nd.array(data, ctx=mx.cpu()))
+            gpu_out = gpu_model(gpu_data)
+        out = cpu_out.asnumpy()
+        max_val = np.max(out)
+        assert_almost_equal(out / max_val, gpu_out.asnumpy() / max_val, rtol=1e-2, atol=1e-2)
+
+def get_nn_model(name):
+    if "densenet" in name:
+        return get_model(name, dropout=0)
+    else:
+        return get_model(name)
+
+def test_training():
+    # We use network models without dropout for testing.
+    # TODO(zhengda) mobilenet can't pass this test even without MKLDNN.
```

Comment:
TODO?

Reply:
I'll look into the problem later. It's not related to this PR. The native MXNet operators can't pass the test with mobilenet. Most likely it's caused by depth-wise convolution.